inspirehep/modules/authors/static/js/authors/author.js

```diff
@@ -0,0 +1,228 @@
+            } else {
+              tempPosition['date'] = start_date + " - " + spaces;
+            }
+          } else if (position.hasOwnProperty('end_date')) {
+            tempPosition['date'] = spaces + " - " + position.end_date;
+          } else {
+            tempPosition['date'] = spaces + " - " + spaces;
+          }
+
+          // Position rank.
+          if (position.hasOwnProperty('rank')) {
+            tempPosition['rank'] = ranks[position.rank];
+
+            if (position.rank == 'MAS' || 
+              position.rank == 'UG') {
+              education.push(tempPosition);
+            } else {
+              work.push(tempPosition);
+            }
+          } else {
+            if (tempPosition.hasOwnProperty('institution') ||
+                tempPosition.hasOwnProperty('rank')) {
+              work.push(tempPosition);
+            }
+          }
+        });
+      }
+
+      var positions = {};
+      positions.education = education;
+      positions.work = work;
+
+      return positions;
+    };
+  }]);
+
+  app.directive('authorAffiliation', function() {
+    return {
+      restrict: 'E',
+      template: '<a href="/search?p=department_acronym:' +
+                '{{ affiliation }}&cc=Institutions">' +
+                '{{ affiliation }}</a>',
+      scope: true,
+      link: function(scope) {
+        if ('positions' in scope.record) {
+          var affiliations = [];
+          var positionsObject = scope.record.positions;
+
+          angular.forEach(positionsObject, function(position) {
+            if ('institution' in position) {
+              if ('name' in position.institution) {
+                affiliations.push(position.institution.name);
+              }
+            }
+          });
+
+          if (affiliations.length > 0) {
+            scope.affiliation = affiliations[0];
+          } else {
+            scope.affiliation = '';
+          }
+        }
+      }
+    };
+  });
+
+  app.directive('authorEmail', function() {
+    return {
+      restrict: 'E',
+      template: '<i class="fa fa-envelope-o"></i> ' +
+                '<a href="mailto:{{ email }}">{{ email }}</a>',
+      scope: true,
+      link: function(scope) {
+        if ('positions' in scope.record) {
+          var emails = [];
+          var emailsObject = scope.record.positions;
+
+          angular.forEach(emailsObject, function(position) {
+            if ('email' in position) {
+              emails.push(position.email);
+            }
+          });
+
+          if (emails.length > 0) {
+            scope.email = emails[0];
+          } else {
+            scope.email = '';
+          }
+        }
+      }
+    };
+  });
+
+  app.directive('authorEducation', function() {
+     return {
+      restrict: 'E',
+      templateUrl: '/static/js/authors/templates/position.html',
+      scope: true,
+      controller: 'AuthorController',
+      link: function(scope) {
+        scope.data = scope.positions.education;
```

Comment:
should it be a more specific attribute inside of `scope`? Instead of using `.data`